### PR TITLE
Inform user about successfully sent receipts

### LINKED ISSUE #6358: Inform user about automatically send receipts
Hide "Send receipt" and display "A receipt has been emailed to [test@test.com](mailto:test@test.com)" when billingEmail is not empty.

This is the current look of the screen.

These are the designs ([figma](https://www.figma.com/file/vaw3DvewbUwuQnPXpdNGGH/Orders---Card-Present-Payments?node-id=2801%3A234067))

### PR DESCRIPTION
**Pull request by staskus**
Closes: #6358

## Description

Show a message explaining the email after the successful payment of the order. This message already exists on Android. 

To satisfy TTP requirements, we need to allow email receipts to be sent after the failed and successful payment. This functionality will be implemented in future tasks. However, we don't want merchants to be misled and send receipts to customers after emails have already been sent automatically.

### Current

We have 4 view models that describe success alert content:

| Class                               | Description                                                                                      | Buttons                                          |
|------------------------------------|--------------------------------------------------------------------------------------------------|-------------------------------------------------|
| `CardPresentModalSuccess`          | Is shown if iOS has a default mail client installed.                                           | `Email Receipt`, `Print Receipt`, `Save receipt and continue` |
| `CardPresentModalBuiltInSuccess`   | Is shown if iOS has a default mail client installed.                                           | `Email Receipt`, `Print Receipt`, `Save receipt and continue` |
| `CardPresentModalSuccessWithoutEmail` | Is shown if iOS doesn't have a default mail client installed.                                 | `Print Receipt`, `Save receipt and continue` |
| `CardPresentModalBuiltInSuccessWithoutEmail` | Is shown if iOS doesn't have a default mail client installed.                           | `Print Receipt`, `Save receipt and continue` |

### New

I expanded the view models with two more options that are shown when payment succeeds and the order has a customer email.

| Class                               | Description                                                                                      | Buttons                                          |
|------------------------------------|--------------------------------------------------------------------------------------------------|-------------------------------------------------|
| `CardPresentModalSuccessEmailSent` | Is shown if successful payment has customer email set.                                         | `Print Receipt`, `Save receipt and continue` |
| `CardPresentModalBuiltInSuccessEmailSent` | Is shown if successful payment has customer email set.                                       | `Print Receipt`, `Save receipt and continue` |

In the future tasks, I plan to introduce more changes:

- `CardPresentModalSuccess` and `CardPresentModalBultInSuccess` cases will be updated to support sending email receipt via the new API.
- Error `CardPresentModalError` and `CardPresentModalNonRetryableError` view models will be expanded in a similar way as success views.

## Steps to reproduce

Configure a site that supports both TTP and Card Reader payments.

### Success Payment with Customer

1. Create order (product or custom amount)
2. Add customer
3. Make TTP or Card Reader payment
4. Confirm that the customer email is shown on the success message

### Success Payment without customer and device with Mail client (Regression)

1. Install and configure mail app on iOS
2. Create order (product or custom amount)
3. Make TTP or Card Reader payment
4. Confirm that three buttons are shown on success modal:  Email Receipt, Print Receipt, Save receipt and Continue

### Success Payment without customer and device with no mail client (Regression)

1. Delete mail app from iOS / use simulator
2. Create order (product or custom amount)
3. Make TTP or Card Reader payment
4. Confirm that two buttons are shown on success modal: Print Receipt, Save receipt and Continue

## Testing information

- Tested legacy local receipt flow. I decided to leave these changes out of the legacy flow for easier maintenance.
- Additionally, I tested the described scenarios on iPad 18.1 as well.

## Screenshots

| TTP | Card Reader |
| --- | --- |
| IMG_9228 | IMG_9230 |

## Android

image

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/232">21.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalSuccess.swift [hunk 1]
```diff
@@ -63,26 +65,36 @@ final class CardPresentModalSuccess: CardPresentPaymentsModalViewModel {
...
-            "Email receipt",
+            "cardPresentPaymentsModal.success.emailReceipt",
+            value: "Email receipt",
             comment: "Button to email receipts. Presented to users after a payment has been successfully collected"
         )
 
         static let saveReceiptAndContinue = NSLocalizedString(
-            "Save receipt and continue",
+            "cardPresentPaymentsModal.success.saveReceiptAndContinue",
+            value: "Save receipt and continue",
```
Comment (joshheald) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalSuccess.swift]: Thanks for updating these to the newer standard

File: WooCommerce/WooCommerceTests/Mocks/MockReceiptEligibilityUseCase.swift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@testable import WooCommerce
+
+final class MockReceiptEligibilityUseCase: ReceiptEligibilityUseCaseProtocol {
+    var isEligibleForBackendReceipts: Bool = false
```
Comment (joshheald) [inline WooCommerce/WooCommerceTests/Mocks/MockReceiptEligibilityUseCase.swift]: Should this default to true? Optimistically, most stores should be eligible, right?

**Review by joshheald — APPROVED** [review]
Works well; I tested on an iPad Air runnind 17.7 and an iPhone running 18.1.

Could you add some release notes before merging please?

File: WooCommerce/WooCommerceTests/Mocks/MockReceiptEligibilityUseCase.swift (hunk 2 above)
Comment (staskus) [reply WooCommerce/WooCommerceTests/Mocks/MockReceiptEligibilityUseCase.swift]: Yes, that's a good point. I'll update it 👍

**Comment by staskus** [comment]
@joshheald, thanks for testing!

> Could you add some release notes before merging please?

I will.

**Pull request merged**